src/Notes/NavigationNudge.php

```diff
@@ -30,6 +30,9 @@ class NavigationNudge {
 	 * Attach hooks.
 	 */
 	public function __construct() {
+		if ( ! Features::exists( 'navigation' ) ) {
```

Comment:
We could potentially run this in the `wc_admin_daily` cron so it's not running on every page load.

https://github.com/woocommerce/woocommerce-admin/blob/cf72d3b23ee637701fc075ef5236201ff2573d3e/src/Events.php#L118

The note would still appear for a day or less when they update, but I don't think that's a big deal.

I'm not sure if we should drop a `NavigationNudge::possibly_remove_note()` in here or simply override the `possibly_add_note()` method to check if removal is required.

Reply:
> But this may be pushing the bounds of this PR. The alternative is to accept that this change won't take effect until after this version of WooCommerce Admin has been packaged in core.

I think that is acceptable. A more comprehensive fix to avoid issues like this going forward should be tackled separately.

Reply:
> We could potentially run this in the `wc_admin_daily` cron so it's not running on every page load.

Definitely.

> The note would still appear for a day or less when they update, but I don't think that's a big deal.

Agreed. That is reasonable, given our current architecture.

> I'm not sure if we should drop a `NavigationNudge::possibly_remove_note()` in here or simply override the `possibly_add_note()` method to check if removal is required.

Reply:
> It would be really nice to have some actions we could hook into before deactivating one of the plugins so we can avoid issues like this going forward.

Would this be something more than the [general deactivation hook support in WordPress](https://developer.wordpress.org/reference/functions/register_deactivation_hook/)?

Reply:
>Would this be something more than the general deactivation hook support in WordPress?

I don't think the plugin actually gets deactivated does it?  It silently sits in the background; we also would want the ability to possibly run something from the unused version prior to returning early in that plugin.

Reply:
>It would be clearer to have it in a possibly_remove_note(). Removing the note in a method called possibly_add_note() is not expected at all.

I agree this is more clear, though the logic for notes being deleted is typically going to be the opposite for adding or deleting and we'll end up with a lot of duplicated logic.  For example:

```php
function possibly_add_note() {
  if ( ! $has_feature ) {
    return;
  }
}
function possibly_delete_note() {
  if ( $has_feature ) {
    return;
  }
}
```

Having a single function might save some headaches down the road if we plan to add/remove a number of notes.  Naming needs work, but something along the lines of `should_have_note()`.

Might be worth saving for a follow-up.